lib/matplotlib/text.py

```diff
@@ -1237,7 +1240,7 @@ def _preprocess_math(self, s):
             if s == " ":
                 s = r"\ "
             return s, "TeX"
-        elif cbook.is_math_text(s):
+        elif self.get_parse_math() and cbook.is_math_text(s):
             return s, True
         else:
             return s.replace(r"\$", "$"), False
```

Comment:
If we aren't passing any math, then we shouldn't unescape `\$`, no?

Reply:
It's what I thought at first, but I didn't want to change the default behaviour (if there wasn't any `parse_math`)

Reply:
I mean, there must be a reason why we do it if the text wasn't interpreted as math right?

Reply:
Because `r'Exchange is CA\$20 for US\$10'` should escape mathtext and turn it off (see `is_math_text`). But if parsing is disabled entirely, we don't need such escapes.